File: Golang/BinaryTreeInorderTraversal/BinaryTreeInorderTraversal_test.go
package BinaryTreeInorderTraversal

import (
	"testing"
	"reflect"
)

func TestInorderTraversal(t *testing.T) {

	if reflect.DeepEqual(inorderTraversal(
		nil),
		[]int{}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(inorderTraversal(
		&TreeNode{
			Val: 1,
			Right: &TreeNode{
				Val: 2,
				Left: &TreeNode{
					Val: 3,
				},
			},
		}),
		[]int{1, 3, 2}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(inorderTraversal(
		&TreeNode{
			Val: 1,
			Left: &TreeNode{
				Val: 2,
				Left: &TreeNode{
					Val: 4,
				},
				Right: &TreeNode{
					Val: 5,
				},
			},
			Right: &TreeNode{
				Val: 3,
				Left: &TreeNode{
					Val: 6,
				},
				Right: &TreeNode{
					Val: 7,
				},
			},
		}),
		[]int{4, 2, 5, 1, 6, 3, 7}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/BinaryTreeInorderTraversalTest.py
from unittest import TestCase

from BinaryTreeInorderTraversal import BinaryTreeInorderTraversal

from BinaryTreeInorderTraversal import TreeNode


class TestBinaryTreeInorderTraversal(TestCase):
    def test_inorderTraversal(self):
        b = BinaryTreeInorderTraversal()

        self.assertTrue(len(b.inorderTraversal(None)) == 0)

        node132 = TreeNode(1)
        right = TreeNode(2)
        right.left = TreeNode(3)
        node132.right = right

        self.assertEqual(b.inorderTraversal(node132), [1, 3, 2])

        node4251637 = TreeNode(1)
        node2 = TreeNode(2)
        node3 = TreeNode(3)
        node2.left = TreeNode(4)
        node2.right = TreeNode(5)
        node3.left = TreeNode(6)
        node3.right = TreeNode(7)
        node4251637.left = node2
        node4251637.right = node3

        self.assertEqual(b.inorderTraversal(node4251637), [4, 2, 5, 1, 6, 3, 7])
